### DISCUSSION TITLE
Are there no comparison functions?

### DISCUSSION THREAD

**Discussion by antoniopgs** [PaulRBerg/prb-math#110] [Ideas]
Is there a `a.greaterThan(b`) or `a.lessThanOrEqualTo(b)`?

**Comment by PaulRBerg** [comment]
No, there aren't any in V2! But there will be comparison functions in V3:

https://github.com/paulrberg/prb-math/blob/11cd8a1b6460dafbca25a0edade0b527cc53862b/contracts/UD60x18.sol#L560-L563

https://github.com/paulrberg/prb-math/blob/11cd8a1b6460dafbca25a0edade0b527cc53862b/contracts/UD60x18.sol#L581-L584

Side note - feel free to take the [`staging`](https://github.com/paulrberg/prb-math/tree/staging) branch for a spin. It should be pretty stable and good to use for now (though, as always, I cannot give warranties).

**Reply by antoniopgs** [reply]
Ok thanks, another thing.
Have you made (or do you know of) ERC20 contracts which use PRBs instead of uints for balances, supply and whatnot?
I can make my own, but just wondering if something is already out there.

**Reply by PaulRBerg** [reply]
I'm sure that somewhere on GitHub there is a project that did just this, but unfortunately I don't know any off the top of my head.

I have something else for you though (sorry for the PRB spam haha!). Take a a look at [PRBContracts](https://github.com/paulrberg/prb-contracts):

https://github.com/paulrberg/prb-contracts/tree/26779ec707aba4347b501f5cba9e00bd2e9c1d3e/src/token/erc20

I basically [wrote](https://twitter.com/PaulRBerg/status/1511692416436391939) a modern implementation of ERC-20 that uses [custom errors](https://blog.soliditylang.org/2021/04/21/custom-errors/) instead of revert reason strings. Custom errors are much cheaper gas-wise.

Though I'm curious - why would you need fixed-point types for ERC-20 balances and supplies?

**Comment by PaulRBerg** [comment]
I have recently released [v3.0.0](https://github.com/paulrberg/prb-math/releases/tag/v3.0.0), which means that PRBMath has comparison functions now 🥳

```solidity
// SPDX-License-Identifier: UNLICENSED
pragma solidity >=0.8.13;

import "@prb/math/UD60x18.sol";

function compare(UD60x18 x, UD60x18 y) pure returns (bool result) {
    result = x.gt(y); // greater than
    result = x.gte(y); // greater than or equal
    result = x.lt(y); // less than
    result = x.lte(y); // less than or equal
}
```
